HereBeDragons-Pins-1.0.lua

```diff
@@ -514,7 +519,11 @@ function pins:AddWorldMapIconWorld(ref, icon, instanceID, x, y)
 
     if WorldMapButton:IsVisible() then
         local currentMapID, currentMapFloor = GetCurrentMapAreaID(), GetCurrentMapDungeonLevel()
-        if currentMapID and HBD.mapData[currentMapID] and HBD.mapData[currentMapID].instance == instanceID then
+        if currentMapID == -1 and GetCurrentMapContinent()==0 then
+            currentMapID = 0  -- force 0=Azeroth
+        end
+        local mapData = HBD.mapData[currentMapID]
+        if currentMapID and mapData and (mapData.instance == instanceID or mapData.floors[instanceID]) then
```

Comment:
Instead of checking mapData.floors[instanceID], I would probabl prefer simply checking for currentMapID == 0, less risk of bad things happening (in the other places that do this as well)